Mathlib/Combinatorics/Digraph/Orientation.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Combinatorics.Digraph.Basic
+import Mathlib.Combinatorics.SimpleGraph.Basic
+
+/-!
+
+# Graph Orientation
+
+This module introduces conversion operations between `Digraph`s and `SimpleGraph`s, by either
+forgetting the edge orientations of `Digraph`, or adding such orientations on a `SimpleGraph`. It
+also introduces oriented graphs.
+
+## Main Definitions
+
+- `Digraph.toSimpleGraphOr`: Converts a `Digraph` to a `SimpleGraph` by creating an undirected edge
+  if either orientation exists in the digraph.
+- `Digraph.toSimpleGraphAnd`: Converts a `Digraph` to a `SimpleGraph` by creating an undirected edge
+  only if both orientations exist in the digraph.
+- `Digraph.IsOriented`: A predicate that determines whether a digraph is oriented (no self-loops and
+  no bidirectional edges).
+- `SimpleGraph.toDigraph`: Converts a `SimpleGraph` to a `Digraph` by creating edges in both
+  directions for each undirected edge.
+
+## TODO
+
+- Show that there is an isomorphism between loopless complete digraphs and oriented graphs.
+- Define more ways to orient a `SimpleGraph`.
+
+## Tags
+
+digraph, simple graph, oriented graphs
+-/
+
+variable {V : Type*}
+
+namespace Digraph
+
+section toSimpleGraph
+
+/-! ### Orientation-forgetting maps on digraphs -/
+
+/--
+Orientation-forgetting map from `Digraph` to `SimpleGraph` that gives an unoriented edge if
+either orientation is present.
+-/
+def toSimpleGraphOr (G : Digraph V) : SimpleGraph V := SimpleGraph.fromRel G.Adj
+
+/--
+Orientation-forgetting map from `Digraph` to `SimpleGraph` that gives an unoriented edge if
+both orientations are present.
+-/
+def toSimpleGraphAnd (G : Digraph V) : SimpleGraph V where
```

Comment:
Do you think you could find more semantic names for these two operations?

Reply:
What do you think of `toSimpleGraphFromEitherEdge` and `toSimpleGraphFromBothEdges`? Or maybe `toSimpleGraphInclusive` and `toSimpleGraphStrict`?

Reply:
Maybe the latter yes